### PR TITLE
pass headers to requests -- fix for #280

### PR DESCRIPTION
**Pull request by ayan-usgs**
Proposed resolution to #280.

Any comments are appreciated.

### CODE REVIEW

File: owslib/util.py [hunk 1]
```diff
@@ -147,7 +147,6 @@ def openURL(url_base, data=None, method='Get', cookies=None, username=None, pass
     Uses requests library but with additional checks for OGC service exceptions and url formatting.
     Also handles cookies and simple user password authentication.
     """
-    headers = {}
```
Comment (tomkralidis) [inline owslib/util.py]: It would be cleaner if headers = None here

File: owslib/util.py [hunk 2]
```diff
@@ -165,14 +164,15 @@ def openURL(url_base, data=None, method='Get', cookies=None, username=None, pass
     if method.lower() == 'post':
         try:
             xml = etree.fromstring(data)
-            headers['Content-Type'] = "text/xml"
+            headers = {'Content-Type': 'text/xml'}
         except (ParseError, UnicodeEncodeError):
-            pass
+            headers = None
```
Comment (tomkralidis) [inline owslib/util.py]: And then removing this line

File: owslib/util.py [hunk 3]
```diff
@@ -165,14 +164,15 @@ def openURL(url_base, data=None, method='Get', cookies=None, username=None, pass
...
 
         rkwargs['data'] = data
 
     elif method.lower() == 'get':
         rkwargs['params'] = data
+        headers = None
```
Comment (tomkralidis) [inline owslib/util.py]: And then removing this line too.

File: owslib/util.py [hunk 4]
```diff
@@ -165,14 +164,15 @@ def openURL(url_base, data=None, method='Get', cookies=None, username=None, pass
     if method.lower() == 'post':
         try:
             xml = etree.fromstring(data)
-            headers['Content-Type'] = "text/xml"
+            headers = {'Content-Type': 'text/xml'}
         except (ParseError, UnicodeEncodeError):
-            pass
```
Comment (tomkralidis) [inline owslib/util.py]: And restoring this line.

**Comment by tomkralidis** [comment]
@ayan-usgs some suggestions dispersed in the code which would make things cleaner and leaner, if possible.

**Comment by ayan-usgs** [comment]
@tomkralidis,

Changes have been made per your suggestions. Please let me know if you have additional comments.

**Comment by tomkralidis** [comment]
Thanks @ayan-usgs!

**Pull request merged**